homeassistant/components/weatherkit/coordinator.py

```diff
@@ -62,10 +63,22 @@ async def _async_update_data(self):
             if not self.supported_data_sets:
                 await self.update_supported_data_sets()
 
-            return await self.client.get_weather_data(
+            updated_data = await self.client.get_weather_data(
```

Comment:
This would be a lot more elegant but I also need to set `self.last_updated_at` to the current time if the fetch is successful, so we can't return immediately. Of course open to suggestions on how to approach it differently.

Reply:
Oh, sorry forgot about that part, you are right, that needs a different approach

Reply:
It is suggesting to return in an else block. Also you should move `self.last_updated_at = datetime.now()` outside of the try block.

```python      
        except WeatherKitApiClientError as exception:
            if self.data is None or (
                self.last_updated_at is not None
                and datetime.now() - self.last_updated_at
                > timedelta(seconds=STALE_DATA_THRESHOLD_SEC)
            ):
                raise UpdateFailed(exception) from exception

            LOGGER.warning("Using stale data because update failed: %s", exception)
            return self.data
        else:
            self.last_updated_at = datetime.now()
            return updated_data
```

Reply:
Ahh, I was unaware Python supported `else` statements on try-catch blocks... that suggestion makes sense now. This looks good, thanks!